airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/record_buffer/InMemoryRecordBufferingStrategy.java

```diff
@@ -91,6 +90,7 @@ public void flushAll() throws Exception {
     }
     close();
     clear();
+    bufferSizeInBytes = 0;
```

Comment:
resets `bufferSizeInBytes` to within `flushAll()` method to prevent issue where after a `BufferedStreamConsumer#periodicBufferFlush` will trigger another buffer flush since the byte count was not reset. This also matches logic held in `SerializedBufferingStrategy` which both implements `BufferingStrategy`

Reply:
Is there any chance the preceding code could fail? wondering what the implications of that would be, like should this be wrapped in a `try/finally` or something?

Reply:
Yes, there's a possibility the two methods could fail. In that case you would not want to reset the `bufferSizeInBytes` so this does not _need_ to be in try/finally block

The most important method that needs to precede the reseting of `bufferSizeInBytes` is the `clear()` method which reassigns the buffer mapping to an empty map